From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 31 Oct 2022 22:21:33 +0100
Subject: [PATCH] Switch to built-in setuptols functionality

https://dustingram.com/articles/2018/03/16/markdown-descriptions-on-pypi/

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/setup.py b/setup.py
index c976e38..46218e6 100644
--- a/setup.py
+++ b/setup.py
@@ -2,17 +2,20 @@
 
 from setuptools import find_packages, setup
 
+with open('README.md') as f:
+    long_description = f.read()
+
 setup(
     name="vvm",
     version="0.1.0",  # don't change this manually, use bumpversion instead
     description="Vyper version management tool",
-    long_description_markdown_filename="README.md",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
     author="Ben Hauser",
     author_email="ben@hauser.id",
     url="https://github.com/vyperlang/vvm",
     include_package_data=True,
     py_modules=["vvm"],
-    setup_requires=["setuptools-markdown"],
     python_requires=">=3.6, <4",
     install_requires=["requests>=2.19.0,<3", "semantic_version>=2.8.1,<3"],
     license="MIT",
