From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 11 Sep 2024 16:49:32 +0200
Subject: [PATCH] Don't try to generate venv (we don't have internet access
 during build)

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/Makefile b/Makefile
index 66d6fe1..1ade49a 100644
--- a/Makefile
+++ b/Makefile
@@ -76,7 +76,6 @@ build-docs-ci:
 
 package-test: clean
 	python -m build
-	python scripts/release/test_package.py
 
 notes: check-bump
 	# Let UPCOMING_VERSION be the version that is used for the current bump
