From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 1 Nov 2022 10:42:37 +0100
Subject: [PATCH] Fedora: use cryptodome explicitly

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/eth_keyfile/keyfile.py b/eth_keyfile/keyfile.py
index 597c98d..f9a9cb2 100644
--- a/eth_keyfile/keyfile.py
+++ b/eth_keyfile/keyfile.py
@@ -18,16 +18,16 @@ from typing import (
 )
 import uuid
 
-from Crypto import (
+from Cryptodome import (
     Random,
 )
-from Crypto.Cipher import (
+from Cryptodome.Cipher import (
     AES,
 )
-from Crypto.Protocol.KDF import (
+from Cryptodome.Protocol.KDF import (
     scrypt,
 )
-from Crypto.Util import (
+from Cryptodome.Util import (
     Counter,
 )
 from eth_keys import (
