From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 5 Nov 2024 12:10:33 +0300
Subject: [PATCH] Adjust tests for hexbytes >= 1.0.0

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/tests/test_abi.py b/tests/test_abi.py
index 04a4118..df07f1b 100644
--- a/tests/test_abi.py
+++ b/tests/test_abi.py
@@ -41,7 +41,7 @@ def test_get_log_topic_for_event_with_dynamic_tuple_size_inputs():
         ],
     }
     result = get_log_topic(event_abi)
-    assert result == "0xa51b9362db3864a3b79228fc175c8781b95be6435ed9b0528c89ec769e033e34"
+    assert result == "a51b9362db3864a3b79228fc175c8781b95be6435ed9b0528c89ec769e033e34"
 
 
 def test_get_log_topic_for_event_with_fixed_tuple_size_inputs():
@@ -59,4 +59,4 @@ def test_get_log_topic_for_event_with_fixed_tuple_size_inputs():
         ],
     }
     result = get_log_topic(event_abi)
-    assert result == "0xa95e22675fbb9c0598f6c3ef02d12b0a8ac59f6b56d9210a5e9f085971384e7a"
+    assert result == "a95e22675fbb9c0598f6c3ef02d12b0a8ac59f6b56d9210a5e9f085971384e7a"
diff --git a/tests/test_decode_event.py b/tests/test_decode_event.py
index 655abea..714995c 100644
--- a/tests/test_decode_event.py
+++ b/tests/test_decode_event.py
@@ -18,7 +18,7 @@ def test_decode_complex_event(complex_log, topic_map):
     event = decode_log(complex_log, topic_map)
     assert event["name"] == "ComplexTypesEvent"
     data = [i["value"] for i in event["data"]]
-    assert data == ["i am a string! such impress", "0x1234567890abcdef"]
+    assert data == ["i am a string! such impress", "1234567890abcdef"]
 
 
 def test_decode_logs(log, complex_log, topic_map):
@@ -47,7 +47,7 @@ def test_decode_logs_no_topic(anon_a_log, topic_map):
         {
             "name": None,
             "address": "0x3194cBDC3dbcd3E11a07892e7bA5c3394048Cc87",
-            "data": "0x00000000000000000000000066ab6d9362d4f35596279692f0251db635165871",
+            "data": "00000000000000000000000066ab6d9362d4f35596279692f0251db635165871",
             "topics": [],
             "decoded": False,
             "logIndex": 0,
diff --git a/tests/test_decode_trace.py b/tests/test_decode_trace.py
index fba76e8..c15aba0 100644
--- a/tests/test_decode_trace.py
+++ b/tests/test_decode_trace.py
@@ -21,7 +21,7 @@ def test_basic(topic_map, raw_trace):
     events = decode_traceTransaction(raw_trace[:400], topic_map)
     assert len(events) == 1
     assert events[0]["name"] == "ComplexTypesEvent"
-    assert [i["value"] for i in events[0]["data"]] == ["hello", "0x6689"]
+    assert [i["value"] for i in events[0]["data"]] == ["hello", "6689"]
 
 
 def test_complex(topic_map, raw_trace):
@@ -33,7 +33,7 @@ def test_complex(topic_map, raw_trace):
         -42,
         "0x31a66f30252cb3983cb4bf10dd6cb9bf67e304d9",
         True,
-        "0x0000000000000000000000000000000000000000000000000000000000001234",
+        "0000000000000000000000000000000000000000000000000000000000001234",
     ]
 
 
