From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 31 May 2023 18:12:01 +0200
Subject: [PATCH] Revert "Remerge master changes"

This reverts commit 62f7d5262e8a37dd3613eddbfda0d253b82316e6.

diff --git a/pyproject.toml b/pyproject.toml
index 1928194..6df7698 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,18 +1,17 @@
 [build-system]
-requires = [
-    "pdm-backend",
-]
+requires = ["pdm-backend"]
 build-backend = "pdm.backend"
 
 [project]
 name = "docopt-ng"
-dynamic = []
+dynamic = ["version"]
 description = "Jazzband-maintained fork of docopt, the humane command line arguments parser."
 authors = [
-    { name = "Nick Crews", email = "nicholas.b.crews@gmail.com" },
+    {name = "Nick Crews", email = "nicholas.b.crews@gmail.com"},
 ]
 dependencies = []
 requires-python = ">=3.7"
+license = {text = "MIT"}
 readme = "README.md"
 classifiers = [
     "Development Status :: 4 - Beta",
@@ -29,11 +28,6 @@ classifiers = [
     "Programming Language :: Python :: Implementation :: CPython",
     "Programming Language :: Python :: Implementation :: PyPy",
 ]
-version = "0.9.0"
-
-[project.license]
-text = "MIT"
-
 [project.urls]
 Homepage = "https://github.com/jazzband/docopt-ng"
 Repository = "https://github.com/jazzband/docopt-ng"
@@ -42,6 +36,7 @@ Repository = "https://github.com/jazzband/docopt-ng"
 source = "file"
 path = "docopt/_version.py"
 
+
 [tool.pdm.dev-dependencies]
 dev = [
     "black",
@@ -53,16 +48,15 @@ dev = [
     "pytest-mypy",
 ]
 
+
 [tool.pytest.ini_options]
-testpaths = [
-    "./tests",
-]
+testpaths = ["./tests"]
 
 [tool.ruff]
 select = [
-    "E",
-    "F",
-    "I",
+    "E",  # pycodestyle
+    "F",  # pyflakes
+    "I",  # isort
 ]
 
 [tool.ruff.isort]
