From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sat, 10 Feb 2024 14:08:07 +0100
Subject: [PATCH] Direct link instead of redirect

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/setup.py b/setup.py
index b51df91..b8f1103 100644
--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,7 @@ setup(
     setup_requires=setup_requirements,
     test_suite="tests",
     tests_require=test_requirements,
-    url="https://github.com/dmuhs/mythx-models",
+    url="https://github.com/Consensys/mythx-models",
     version="2.2.0",
     zip_safe=False,
 )
