From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 11 Sep 2024 20:23:15 +0200
Subject: [PATCH] Disable this test (we don't have internet access anyway)

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/scripts/release/test_package.py b/scripts/release/test_package.py
index 2f23898..dcf529b 100644
--- a/scripts/release/test_package.py
+++ b/scripts/release/test_package.py
@@ -34,17 +34,3 @@ def install_wheel(venv_path: Path, wheel_path: Path) -> None:
         [venv_path / "bin" / "pip", "install", f"{wheel_path}"],
         check=True,
     )
-
-
-def test_install_local_wheel() -> None:
-    with TemporaryDirectory() as tmpdir:
-        venv_path = create_venv(Path(tmpdir))
-        wheel_path = find_wheel(Path("."))
-        install_wheel(venv_path, wheel_path)
-        print("Installed", wheel_path.absolute(), "to", venv_path)
-        print(f"Activate with `source {venv_path}/bin/activate`")
-        input("Press enter when the test has completed. The directory will be deleted.")
-
-
-if __name__ == "__main__":
-    test_install_local_wheel()
