From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 31 May 2023 19:14:03 +0200
Subject: [PATCH] Replace unmaintained docopt

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/setup.cfg b/setup.cfg
index 1a95495..3334b22 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,5 +1,5 @@
 [metadata]
-name = docopt-ng
+name = docopt
 version = attr: docopt.__version__
 url = https://github.com/jazzband/docopt-ng
 license = MIT
