From 59cbca5039904b6559029f4710ef8a31e5214565 Mon Sep 17 00:00:00 2001
From: swayson <swayson1@gmail.com>
Date: Thu, 17 Sep 2020 12:38:33 +0200
Subject: [PATCH 2/2] Update the setup and tox settings to pytest

---
 setup.py | 5 +++--
 tox.ini  | 4 ++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index cee63dd..399ceac 100644
--- a/setup.py
+++ b/setup.py
@@ -21,8 +21,9 @@ setup(
     author_email='erikrose@grinchcentral.com',
     license='MIT',
     packages=find_packages(exclude=['ez_setup']),
-    tests_require=['nose'],
-    test_suite='nose.collector',
+    setup_requires=['pytest-runner'],
+    tests_require=['pytest'],
+    test_suite='tests',
     url='https://github.com/erikrose/parsimonious',
     include_package_data=True,
     install_requires=['six>=1.9.0'],
diff --git a/tox.ini b/tox.ini
index 9a8519a..00db0de 100644
--- a/tox.ini
+++ b/tox.ini
@@ -10,7 +10,7 @@ envlist = py27, py33, py34, py35, py36
 
 [testenv]
 usedevelop = True
-commands = nosetests parsimonious
+commands = py.test
 deps =
-  nose
+  pytest
   six>=1.9.0
-- 
2.33.1

