From 4810b412f50be70cb70f60d1cee546c43249826b Mon Sep 17 00:00:00 2001
From: pacrob <5199899+pacrob@users.noreply.github.com>
Date: Tue, 19 Mar 2024 10:14:21 -0600
Subject: [PATCH] bump hexbytes to >=1, and eth-rlp to >=2 which has the same
 hexbytes dep


diff --git a/setup.py b/setup.py
index 4ac6665..b6c8302 100644
--- a/setup.py
+++ b/setup.py
@@ -52,9 +52,9 @@ setup(
         "eth-abi>=4.0.0-b.2",
         "eth-keyfile>=0.6.0",
         "eth-keys>=0.4.0",
-        "eth-rlp>=0.3.0",
+        "eth-rlp>=2.0.0",
         "eth-utils>=2.0.0",
-        "hexbytes>=0.1.0,<0.4.0",
+        "hexbytes>=1.0.0",
         "rlp>=1.0.0",
     ],
     python_requires=">=3.8, <4",
