From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 14 Dec 2022 18:01:32 +0100
Subject: [PATCH] Revert "Revert "Support new z3 AST node""

This reverts commit a4d2d0d775463214172a6a531527358d3e5e4642.

diff --git a/libsmtutil/Z3Interface.cpp b/libsmtutil/Z3Interface.cpp
index 962898863..4abeffa16 100644
--- a/libsmtutil/Z3Interface.cpp
+++ b/libsmtutil/Z3Interface.cpp
@@ -302,6 +302,7 @@ Expression Z3Interface::fromZ3Expr(z3::expr const& _expr)
 		arguments.push_back(fromZ3Expr(_expr.arg(i)));
 
 	auto kind = _expr.decl().decl_kind();
+
 	if (_expr.is_ite())
 		return Expression::ite(arguments[0], arguments[1], arguments[2]);
 	else if (_expr.is_not())
@@ -375,7 +376,10 @@ Expression Z3Interface::fromZ3Expr(z3::expr const& _expr)
 		return Expression("dt_accessor_" + _expr.decl().name().str(), arguments, sort);
 	else if (kind == Z3_OP_DT_IS)
 		return Expression("dt_is", {arguments.at(0)}, sort);
-	else if (kind == Z3_OP_UNINTERPRETED)
+	else if (
+		kind == Z3_OP_UNINTERPRETED ||
+		kind == Z3_OP_RECURSIVE
+	)
 		return Expression(_expr.decl().name().str(), arguments, fromZ3Sort(_expr.get_sort()));
 
 	smtAssert(false);
