From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sun, 25 Dec 2022 22:43:26 +0100
Subject: [PATCH] Workaround for compiler's issue

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/libsolidity/ast/ASTJsonExporter.cpp b/libsolidity/ast/ASTJsonExporter.cpp
index d308cd90c..5d0584e8d 100644
--- a/libsolidity/ast/ASTJsonExporter.cpp
+++ b/libsolidity/ast/ASTJsonExporter.cpp
@@ -138,7 +138,8 @@ Json::Value ASTJsonExporter::sourceLocationsToJson(vector<SourceLocation> const&
 
 string ASTJsonExporter::namePathToString(std::vector<ASTString> const& _namePath)
 {
-	return boost::algorithm::join(_namePath, ".");
+	std::string sep = ".";
+	return boost::algorithm::join(_namePath, sep);
 }
 
 Json::Value ASTJsonExporter::typePointerToJson(Type const* _tp, bool _withoutDataLocation)
diff --git a/libsolidity/codegen/ir/IRGeneratorForStatements.cpp b/libsolidity/codegen/ir/IRGeneratorForStatements.cpp
index 61ec2b144..2f487fdeb 100644
--- a/libsolidity/codegen/ir/IRGeneratorForStatements.cpp
+++ b/libsolidity/codegen/ir/IRGeneratorForStatements.cpp
@@ -179,7 +179,7 @@ private:
 			else
 			{
 				solAssert(!IRVariable{*varDecl}.hasPart("offset"));
-				value = "0";
+				value = string("0");
 			}
 		}
 		else if (varDecl->type()->dataStoredIn(DataLocation::CallData))
