From: Peter Lemenkov <lemenkov@gmail.com>
Date: Fri, 1 Jul 2022 12:09:08 +0200
Subject: [PATCH] Initialize vars before use

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/liblangutil/Scanner.cpp b/liblangutil/Scanner.cpp
index d769a9363..1b9b2afac 100644
--- a/liblangutil/Scanner.cpp
+++ b/liblangutil/Scanner.cpp
@@ -540,8 +540,8 @@ void Scanner::scanToken()
 
 	Token token;
 	// M and N are for the purposes of grabbing different type sizes
-	unsigned m;
-	unsigned n;
+	unsigned m = 0;
+	unsigned n = 0;
 	do
 	{
 		// Remember the position of the next token
