From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 10 Apr 2023 15:40:39 +0200
Subject: [PATCH] Ease version requirements

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/setup.py b/setup.py
index 8fb5d2e9..b73e0084 100644
--- a/setup.py
+++ b/setup.py
@@ -94,7 +94,7 @@ setup(
     install_requires=[
         "asttokens>=2.0.5,<3",
         "pycryptodomex>=3.5.1,<4",
-        "semantic-version>=2.10,<3",
+        "semantic-version>=2.8,<3",
         "importlib-metadata",
         "wheel",
     ],
