From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 1 Nov 2022 20:45:19 +0100
Subject: [PATCH] Use Cryptodomex

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/setup.py b/setup.py
index 05cb5225..8fb5d2e9 100644
--- a/setup.py
+++ b/setup.py
@@ -93,7 +93,7 @@ setup(
     py_modules=["vyper"],
     install_requires=[
         "asttokens>=2.0.5,<3",
-        "pycryptodome>=3.5.1,<4",
+        "pycryptodomex>=3.5.1,<4",
         "semantic-version>=2.10,<3",
         "importlib-metadata",
         "wheel",
diff --git a/vyper/utils.py b/vyper/utils.py
index 2440117d..5fab48df 100644
--- a/vyper/utils.py
+++ b/vyper/utils.py
@@ -42,7 +42,7 @@ decimal.setcontext(DecimalContextOverride(prec=78))
 
 
 try:
-    from Crypto.Hash import keccak  # type: ignore
+    from Cryptodome.Hash import keccak  # type: ignore
 
     keccak256 = lambda x: keccak.new(digest_bits=256, data=x).digest()  # noqa: E731
 except ImportError:
