From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 1 Nov 2022 20:45:19 +0100
Subject: [PATCH] Use Cryptodome

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/vyper/utils.py b/vyper/utils.py
index 90b0675c..eb13801a 100644
--- a/vyper/utils.py
+++ b/vyper/utils.py
@@ -29,7 +29,7 @@ decimal.setcontext(DecimalContextOverride(prec=78))
 
 
 try:
-    from Crypto.Hash import keccak  # type: ignore
+    from Cryptodome.Hash import keccak  # type: ignore
 
     keccak256 = lambda x: keccak.new(digest_bits=256, data=x).digest()  # noqa: E731
 except ImportError:
